app/browser/api/ledger.js

```diff
@@ -2243,9 +2243,22 @@ const run = (state, delayTime) => {
     return
   }
 
+  const publishers = ledgerState.getPublisher(state)
+  if (publishers.isEmpty()) {
+    const newReconcileTime = (new Date().getTime() + (ledgerUtil.milliseconds.day * 30))
+    client.setTimeUntilReconcile(newReconcileTime, (err, stateResult) => {
```

Comment:
Let's extract this block into separate function and just call this function here. Also please replace this block in other places that you can find it with this new function. You don't need to pass newReconcileTime, you can just pass `null ` in, which will default to 30 days.